src/tools/snapshot/nodes/BlockSerializationNode.java

```diff
@@ -57,7 +57,7 @@ public void serialize(final SBlock block, final SnapshotBuffer sb) {
 
       for (int i = 0; i < args.length; i++) {
         Types.getClassOf(args[i]).serialize(args[i], sb);
-        sb.putLongAt(base + (i * Long.BYTES), sb.getObjectPointer(args[i]));
+        sb.putLongAt(base + (i * Long.BYTES), sb.getRecord().getObjectPointer(args[i]));
```

Comment:
The `getRecord()` is this performance critical? Does it need to be moved out of the loop?
I mean, all the code here is unoptimized, for instance the `Types.getClassOf` isn't even going to compile.

Reply:
I moved it out of the loop, it's a simple getter. I'm adding a todo for optimisation, first of all we need to specialise on the different blocks, then we can cache the argument serialisation.